app/src/main/java/au/id/neasbey/biblecast/util/BibleAPIRequest.java

```diff
@@ -0,0 +1,177 @@
+package au.id.neasbey.biblecast.util;
+
+import android.text.Spanned;
+import android.text.TextUtils;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by craigneasbey on 8/07/15.
+ *
+ * Bible API request handling
+ */
+public class BibleAPIRequest {
+
+    private static final String TAG = BibleAPIRequest.class.getSimpleName();
+
+    private String requestUrl = "";
```

Comment:
This class seems to have a few responsibilities ... building the request url, making a the request and handling the response. These responsibilities have been moved into private methods which is nice, but I'd like to see each responsibility moved into its own (unit testable) class.